setup.py

```diff
@@ -24,19 +25,30 @@ def read_requirements():
 
 
 def read_extra_requirements():
-
-    extra_requirements = {}
-    extra_requirements["all"] = []
+    extra_requirements = defaultdict(lambda: [])
+    collect_key = "all"
     with open("./extra-requirements.txt") as f:
         for req in f:
             req = req.replace("\n", " ")
-            extra_requirements[_cut_version_number_from_requirement(req)] = [req]
-            extra_requirements["all"].append(req)
+            if req.startswith("#"):
```

Comment:
i think this logic here is a bit misleading. why not really put all reqs in the key "all", and only consider the other two ("dev", "doc") separately. alternatively, maybe use "extra" for all except "dev" and "doc"? using "all" but excluding "dev", "doc" confused me

Reply:
`all` here means "all requirements necessary for using the library as a user". But okay, we can use `extra` for that purpose and then `all` for everything.

All your other comments are already outdated because I fixed the logic in the travis file.

Reply:
Done in 93f2adcf19e319ce8ee98798e83b296116e4878d